### ISSUE TITLE
There are some problems when using the [FragmentNavigator.TransactionType.ADD], Why?

### ISSUE THREAD

**Issue by xuexixuexijpg** [KaustubhPatange/navigator#23]
Hello, there are some problems here, as shown below:  
After using `add`, the expectation should be the `show/hide` operation of the `fragment`, right? But they overlap and can be clicked, why?

![image](https://user-images.githubusercontent.com/57759182/170466205-539ac76b-b461-429b-a12a-6a2fb9b43d2f.png)

**Comment by xuexixuexijpg** [comment]
I try to use hide method to `hide `the original `fragment `so that the fragments don't overlap? But how to get the last fragment of the navigation?

![image](https://user-images.githubusercontent.com/57759182/170503134-f567d003-7ed8-44d7-8a36-e466a8eef991.png)

**Comment by KaustubhPatange** [comment]
They will overlap when you use `TransactionType.ADD`. It is similar to,

```kotlin
fragmentManager.beginTransaction()
    .add(...)
    .commit()
```
So if in your container there is an existing fragment, they will overlap. Keep in mind that `add` means it will keep two fragments actively in a container. This should only be used when you know what you are doing & must be avoided in other cases.

Now coming to your problem, I don't know what you are trying to do by applying `when` clause to `NavOptions.transaction` because it seems you are directly calling a fragment transaction instead of `FragmentNavigator.navigateTo` call. What are you implying here?

I'm guessing you want to hide the previous fragment right? In that case since `FragmentNavigator.navigateTo` is an asynchronous commit, it will run at a later period so if you want to hide you should use `commitNow()` before navigating. Also it seems like since you are in a multi-module the navigator associated with `parentNavigator` is different, that is not the `FragmentNavigator` used for transaction. In my opinion, you should try to understand how the multi-module sample works so you will understand how navigation works in a multi-module. 

Also, I don't think I've ever exposed a `history.getContents()` method, did you add it? Because I'm seeing that it is returning an empty list. If you want the current active fragment in the container use `FragmentNavigator.getCurrentFragment()`.

Now simply, if you want still want to remove the overlay, set the background on the layout file. Typically it inherits `?attr/windowBackground`.

**Comment by xuexixuexijpg** [comment]
> They will overlap when you use `TransactionType.ADD`. It is similar to,
> 
> ```kotlin
> fragmentManager.beginTransaction()
>     .add(...)
>     .commit()
> ```
> 
> So if in your container there is an existing fragment, they will overlap. Keep in mind that `add` means it will keep two fragments actively in a container. This should only be used when you know what you are doing & must be avoided in other cases.
> 
> Now coming to your problem, I don't know what you are trying to do by applying `when` clause to `NavOptions.transaction` because it seems you are directly calling a fragment transaction instead of `FragmentNavigator.navigateTo` call. What are you implying here?
> 
> I'm guessing you want to hide the previous fragment right? In that case since `FragmentNavigator.navigateTo` is an asynchronous commit, it will run at a later period so if you want to hide you should use `commitNow()` before navigating. Also it seems like since you are in a multi-module the navigator associated with `parentNavigator` is different, that is not the `FragmentNavigator` used for transaction. In my opinion, you should try to understand how the multi-module sample works so you will understand how navigation works in a multi-module.
> 
> Also, I don't think I've ever exposed a `history.getContents()` method, did you add it? Because I'm seeing that it is returning an empty list. If you want the current active fragment in the container use `FragmentNavigator.getCurrentFragment()`.
> 
> Now simply, if you want still want to remove the overlay, set the background on the layout file. Typically it inherits `?attr/windowBackground`.

Yes, when using `add`, I want to hide the previous fragment, not `replace`, that is the method I added in your frame, and if it is not hidden, the overlapping of two fragments will be clickable on both, as if adding a background The color is useless, so I added some logic in it. I changed it below. The version I am currently using cannot meet my expectations. I may have to copy a copy of your library to modify it.

![image](https://user-images.githubusercontent.com/57759182/170513278-6a03d669-60ef-44e2-b79f-fca861324e88.png)

**Comment by KaustubhPatange** [comment]
You don't need to copy the library & meet your expectation, the library is just an abstraction over the existing `FragmentManager`'s API. If you know how to use them you don't even need this library. Also, this might be an issue with the fragment container's z index use, `FragmentNavigator.autoChildElevation()` extension function.

Anyway, I suggest you take a look at [XClipper](https://github.com/KaustubhPatange/XClipper/tree/master/XClipper.Android) & [Moviesy](https://github.com/KaustubhPatange/Moviesy) application which uses navigator as their navigation library you will learn how to use it effectively. I'm saying this because what you are doing & trying to achieve is completely wrong. You don't even need to hide the fragments.

**Comment by xuexixuexijpg** [comment]
Yes, this library is abstract, but I can't directly call the abstract methods when I call it outside. I wrote some more code, so I want to change it on the original base library, maybe I can add a `TransactionType `to the enum `TransactionType. ADD_AND_HIDE`?
My current practice project refers to a lot of multi-module practices, although I haven't used compose yet. In terms of routing, it is cumbersome to use xml for `Jetpack Navigation`, and some Ali `ARouter `and Meituan `WMRouter `are also a bit cumbersome, so I found a correct one. Fragment management abstract library, your abstract library is quite suitable, in short, thank you very much for your suggestions and help!

**Comment by KaustubhPatange** [comment]
I would recommend you to read this article thoroughly https://proandroiddev.com/scalable-navigation-in-multi-module-projects-c3c44dc1b913. It has great content on how to effectively deal navigation in multi-module projects.

You may not get this article when you read it once but trust me it will click once you know that the real key in multi-module navigation is basically DI (dagger or hilt) & not a navigation library i.e you can achieve multi-module even with fragment manager APIs.
